### ISSUE TITLE
Work around SQLite's error: HAVING clause on a non-aggregate query

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#14239]
This should be valid in most RDBMS (assuming a dummy table to add a possibly mandatory `FROM` clause):

```sql
select 1 having true
```

In the SQL standard, `HAVING` without `GROUP BY` generates an implicit `GROUP BY ()` using the empty grouping set. But we get:

> SQL Error [1]: [SQLITE_ERROR] SQL error or missing database (HAVING clause on a non-aggregate query)

Unfortunately, SQLite dosen't support grouping sets:

```sql
select 1 group by () having true
```

Produces:

> SQL Error [1]: [SQLITE_ERROR] SQL error or missing database (near ")": syntax error)

But we already emulate it like this:

```sql
select 1 group by (select 1) having true
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.6 (https://github.com/jOOQ/jOOQ/issues/14240), and 3.16.12 (https://github.com/jOOQ/jOOQ/issues/14241)

**Comment by lukaseder** [comment]
BigQuery is also affected
